domain/schema/model.go

```diff
@@ -325,3 +341,99 @@ CREATE UNIQUE INDEX idx_unit_net_node
 ON unit (net_node_uuid);
 `)
 }
+
+func storageSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE storage_pool (
+    uuid TEXT PRIMARY KEY,
+    name TEXT NOT NULL,
+    -- TODO (manadart 2023-11-29): Storage pools in Mongo were stored as settings.
+    -- The types are sourced from the provider.
+    -- We can:
+    --   1. Leave it as text on the basis that we don't know up front what types the provider offers.
+    --   2. Make a lookup with all the known types, which would need maintenance over the long run.
+    --   3. Populate a lookup at model creation with the provider's known types.
```

Comment:
This also has the same maintenance issue as option 2, as a juju upgrade could result in a given provider offering additional types.
Also, we do have 3 known out of the box types, common to all providers. Would be nice to have a look up for those, but unknown provider types complicates it all.

Reply:
Keeping the status quo for now. Updated the comment.